nrf_802154/serialization/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+# Platform needs paths to serialization headers
+target_include_directories(nrf-802154-platform
```

Comment:
All public and pvirate header files seem to be mixed in two directories now. I can't tell which public headers should be included in host build, which in remote build, and which describe platform API.

I remember in one of private repositories we had it clearly separated. Could we create such separation here?